R/mediationanalysis.R

```diff
@@ -602,11 +602,13 @@ MediationAnalysis <- function(jaspResults, dataset, options, ...) {
   # Create semplot model and unv the names of the manifest variables
   # Sorry, this code is really ugly but all it does is replace names for plot.
   semPlotMod <- semPlot::semPlotModel(list(lavResult), list(mplusStd = "std"))[[1]]
-
+  
   manifests <- semPlotMod@Vars$name[semPlotMod@Vars$manifest]
   semPlotMod@Vars$name[semPlotMod@Vars$manifest] <- decodeColNames(manifests)
   semPlotMod@Pars$lhs <- vapply(semPlotMod@Pars$lhs, function(v) ifelse(nchar(v) > 0, decodeColNames(v), ""), "")
   semPlotMod@Pars$rhs <- vapply(semPlotMod@Pars$rhs, function(v) ifelse(nchar(v) > 0, decodeColNames(v), ""), "")
+  if(.hasSlot(semPlotMod, "Thresholds"))
+    semPlotMod@Thresholds$lhs <- vapply(semPlotMod@Thresholds$lhs, function(v) ifelse(nchar(v) > 0, decodeColNames(v), ""), "")
```

Comment:
So I don't understand the code well enough (darned S4 classes) but what types are `semPlotMod@Pars$rhs` and `semPlotMod@Thresholds$lhs`? If they are lists, then what you're doing is fine. If they are character vectors, then the `vapply` is redundant.

Reply:
Yeah good question. I mindlessly copied the approach from Erik-Jan for the other slots in the object. I think all of them are just columns in a dataframe, so you are right that all the `vapply`s in the code here are redundant. I will check whether that's the case and simplify if possible.